utils/frame/benchmarking-cli/src/storage/write.rs

```diff
@@ -62,6 +62,23 @@ impl StorageCmd {
 		let mut rng = Self::setup_rng();
 		kvs.shuffle(&mut rng);
 
+		// Run some rounds of the benchmark as warmup.
+		// The code here is a copy of the instrumentation run below, look there for comments.
+		for i in 0..self.params.warmups {
+			info!("Warmup round {}/{}", i + 1, self.params.warmups);
+			for (k, original_v) in kvs.clone().iter() {
+				let mut new_v = vec![0; original_v.len()];
+				rng.fill_bytes(&mut new_v[..]);
+
+				let replace = vec![(k.as_ref(), Some(new_v.as_ref()))];
+				let (_, stx) = trie.storage_root(replace.iter().cloned(), self.state_version());
+				let tx = convert_tx::<Block>(stx.clone(), true, state_col, supports_rc);
```

Comment:
I would not run the commits and convert_tx for warmup: IIUC warm up is to feed cache.
@ggwpez couldn't we just run the same warmup as for read?

Reply:
Hm yea I am not sure what to best do here.  
Committing like this is quite slow, so if you think just using the same thing as in the read benchmark is good then let's do that.

Reply:
I've pushed the other changes, will take a look at updating these warmups when I get time!

Reply:
(Unfortunately I'm not yet very familiar with benchmarking or warmups so I may be missing something obvious)

I've removed the `convert_tx` and commits, but didn't manage to use the same warmup as for read, since the `client`s used in read and write don't share the same `StorageProvider::storage` trait bound, and I don't see anything equivalent in the write `client`. What do you think about this version?

Alternatively, we can simplify the write benchmark warmup a step compared to read to become:
```rust
for i in 0..self.params.warmups {
    info!("Warmup round {}/{}", i + 1, self.params.warmups);
    let _ = client.header(block)
        .expect("Checked above to exist")
        .ok_or("Value unexpectedly empty")?;
}
```

Reply:
I am not really sure about what is needed for warmup.

But, you can probably add the bound defined in read.rs to write.rs (StorageProvider) and use client parameter, then use the exact same code as in read.
Otherwhise logic from read could be the same with the trie (collect all key using trie iter, then random and use trie get on it), but I think adding the bound will be harmless (both read and write probably use the same client parameter when called).

Reply:
To avoid code duplication, I move the benchmark up one level since we want to call the same code for both read & write. Will this approach work or should the warmups be "closer" to the actual benchmark code?

Reply:
This works for me. I think we don't need to make it more complicated for this iteration.  
What do you think @cheme?

Reply:
looks good